src/UI/Art/SpriteArtist/ControlOptionsSetupArtist.java

```diff
@@ -0,0 +1,177 @@
+package UI.Art.SpriteArtist;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+
+import UI.InputHandler;
+import UI.InputHandler.InputAction;
+
+public class ControlOptionsSetupArtist
+{
+  private static InputHandler myControl = null;
+
+  static ArrayList<KeyBindingPanel> kbps;
+  static {
+    kbps = new ArrayList<KeyBindingPanel>();
+    for( InputAction ia : InputHandler.InputAction.values() )
+    {
+      kbps.add(new KeyBindingPanel(ia));
+    }
+  }
+
+  private static SpriteCursor spriteCursor = new SpriteCursor();
+
+  public static void draw(Graphics g, InputHandler control)
+  {
+    // Draw a fancy background.
+    DiagonalBlindsBG.draw(g);
+
+    myControl = control;
+
+    // Set up some initial parameters.
+    int spacing = kbps.get(0).getImage().getHeight()*2;
+    int xDraw = spacing;
+    int yDraw = spacing/2;
+
+    // Find the selected command/key.
+    int selectedAction = myControl.actionCommandSelector.getSelectionNormalized();
+    int selectedKey = myControl.getKeySelector(InputHandler.InputAction.values()[selectedAction]).getSelectionNormalized();
+
+    // Create an un-scaled image to draw everything at real size before scaling it to the screen.
+    Dimension dimensions = SpriteOptions.getScreenDimensions();
+    int drawScale = SpriteOptions.getDrawScale();
+    BufferedImage controlsImage = SpriteLibrary.createTransparentSprite(dimensions.width/drawScale, dimensions.height/drawScale);
+    Graphics cig = controlsImage.getGraphics();
+
+    // Loop through and draw everything, building or updating the necessary images.
+    for( int ip = 0; ip < kbps.size(); ++ip )
+    {
+      KeyBindingPanel kbp = kbps.get(ip);
+      boolean drawAltAdd = (ip == selectedAction) && myControl.isAssigningKey();
+      BufferedImage panel = kbp.getImage(drawAltAdd);
+      cig.drawImage(panel, xDraw, yDraw, null);
+
+      if( ip == selectedAction ) // Draw the cursor over the selected item.
+      {
+        kbp.setCursorLocationRelative(spriteCursor, selectedKey, xDraw, yDraw);
+        spriteCursor.draw(cig);
+      }
+
+      yDraw += spacing;
+    }
+
+    // Redraw to the screen at scale.
+    g.drawImage(controlsImage, 0, 0, controlsImage.getWidth()*drawScale, controlsImage.getHeight()*drawScale, null);
+  }
+
+  /**
+   * Renders itself into an image like this, with no scaling applied.
+   * +---------+  +------+  +------+  +-----+
+   * | Command |  | Key1 |  | Key2 |  | Add |
+   * +---------+  +------+  +------+  +-----+
+   */
+  private static class KeyBindingPanel
+  {
+    InputHandler.InputAction myAction;
+
+    // A couple of helper quantities.
+    private static final int textBufferPx = 4;
+    private static final int keyBufferPx = 8;
+
+    // The composed TeamPanel image.
```

Comment:
A team of keybindings? :stuck_out_tongue:

Reply:
Of course. A single bound key wouldn't be worth a whole lot in this game.